pywinauto/controls/BaseWrapper.py

```diff
@@ -0,0 +1,816 @@
+        for win_class in cls.windowclasses:
+            _MetaWrapper.re_wrappers[re.compile(win_class)] = cls
+            _MetaWrapper.str_wrappers[win_class] = cls
+        for control_type in cls.controltypes:
+            _MetaWrapper.control_types[control_type] = cls
+
+    @staticmethod
+    def FindWrapper(element):
+        "Find the correct wrapper for this native element"
+        if isinstance(element, six.integer_types):
+            from ..NativeElementInfo import NativeElementInfo
+            element = NativeElementInfo(element)
+        class_name = element.className
+
+        try:
+            return _MetaWrapper.str_wrappers[class_name]
+        except KeyError:
+            wrapper_match = None
+
+            for regex, wrapper in _MetaWrapper.re_wrappers.items():
+                if regex.match(class_name):
+                    wrapper_match = wrapper
+                    _MetaWrapper.str_wrappers[class_name] = wrapper
+
+                    return wrapper
+
+        # if it is a dialog then override the wrapper we found
+        # and make it a DialogWrapper
+        if handleprops.is_toplevel_window(element.handle):
+            from . import win32_controls
+            wrapper_match = win32_controls.DialogWrapper
+
+        if wrapper_match is None:
+            from .HwndWrapper import HwndWrapper
+            wrapper_match = HwndWrapper
+        return wrapper_match
+
+    @staticmethod
+    def FindWrapperUIA(elementinfo):
+        "Find the wrapper for this elementinfo"
+        if isinstance(elementinfo, six.integer_types):
+            from ..UIAElementInfo import UIAElementInfo
+            elementinfo = UIAElementInfo(elementinfo)
+
+        if elementinfo.handle is not None:
+            wrapper = _MetaWrapper.FindWrapper(elementinfo)
+
+            from .HwndWrapper import HwndWrapper
+            if wrapper == HwndWrapper:
+                if elementinfo.controlType in _MetaWrapper.control_types.keys():
+                    wrapper = _MetaWrapper.control_types[elementinfo.controlType]
+        else:
+            # TODO: temporary thing (there is no UIA based wrappers tree yet)
+            from .UIAWrapper import UIAWrapper
+            wrapper = UIAWrapper
+
+        return wrapper
+
+#=========================================================================
+@six.add_metaclass(_MetaWrapper)
+class BaseWrapper(object):
+    """
+    Abstract wrapper for elements.
+
+    All other wrappers are derived from this.
+    """
+
+    # Properties required for _MetaWrapper class
+    friendlyclassname = None
+    windowclasses = []
+    controltypes = []
+
+    # Properties that describe type of the element
+    can_be_label = False
+    has_title = True
+
+    #------------------------------------------------------------
+    def __new__(cls, elementInfo):
+        # only use the meta class to find the wrapper for BaseWrapper
+        # so allow users to force the wrapper if they want
+        if cls != BaseWrapper:
```

Comment:
```if cls is not BaseWrapper:``` is a bit cleaner (some say it even faster) 
http://stackoverflow.com/questions/132988/is-there-a-difference-between-and-is-in-python
http://jaredgrubb.blogspot.co.nz/2009/04/python-is-none-vs-none.html